wordpress-mobile/WordPress-iOS
# Add Noticons font to content and service extensions

Fixes a crash which occurs in all builds after 7b1a177fc82ebe080fa08ff1dc164bd2b7f2ace2. The crash does not seem to appear in Sentry – maybe a lack of crash logging from extensions?

The following log came from the 17.1 release build from the Xcode Organizer's Crash reports (it is the top item in that list):
```
Thread 0 Crashed:
0   ...otificationServiceExtension	0x00000001028b8f90 Swift runtime failure: Unexpectedly found nil while unwrapping an Optional value + 0 (RemoteNotificationStyles.swift:29)
1   ...otificationServiceExtension	0x00000001028b8f90 closure #1 in RemoteNotificationStyles.noticonFont.getter + 0 (RemoteNotificationStyles.swift:29)
2   ...otificationServiceExtension	0x00000001028b8f90 RemoteNotificationStyles.noticonFont.getter + 220 (RemoteNotificationStyles.swift:27)
```

Unsure of the actual impact of this crash but it looks like we've seen issues with notification display in the past with similar problems: https://github.com/wordpress-mobile/WordPress-iOS/issues/13412

## Testing

* Build and run the WordPress app.
* Ensure that the Noticons font continues to display.
* Build and run the `WordPressNotificationServiceExtension`

**Note:** I was unable to reproduce this crash and unable to debug the Service Extension from Xcode. If you have any luck or ideas, let me know!
* I attempted to build and run the `WordPressNotificationServiceExtension` target and wasn't able to get the debugger to attach.
* I attempted to attach to the `WordPressNotificationServiceExtension` process name and drag and drop a test apns file. The test apns did trigger a notification but the debugger never attached to the process and I'm not sure that the extension was actually launched.

<details>
  <summary>Test APNS payload</summary>

```json
{
  "Simulator Target Bundle": "org.wordpress",
  "aps": {
    "badge": 2,
    "alert": {
      "title": "Test",
      "body": "Hello"
    },
    "sound": "default",
    "mutable-content": 1
  }
}
```

</details>

## Regression Notes
1. Potential unintended areas of impact

Rich notification display (those with `mutable-content: 1`)

2. What I did to test those areas of impact (or what existing automated tests I relied on)

I used a test apns file which was properly delivered before and after change.

3. What automated tests I added (or what prevented me from doing so)

Did not add any automated tests but I wondered if we could add something to check target membership at build time so we aren't caught by surprise.

PR submission checklist:
